File: apiwdbe/WdbeQCmdARetpar.cpp
/**
	* \file WdbeQCmdARetpar.cpp
	* API code for table TblWdbeQCmdARetpar (implementation)
	* \copyright (C) 2016-2020 MPSI Technologies GmbH
	* \author Alexander Wirthmueller (auto-generation)
	* \date created: 5 Dec 2020
  */
// IP header --- ABOVE

#include "WdbeQCmdARetpar.h"

using namespace std;
using namespace Sbecore;
using namespace Xmlio;

/******************************************************************************
 class WdbeQCmdARetpar
 ******************************************************************************/

WdbeQCmdARetpar::WdbeQCmdARetpar(
			const uint jnum
			, const string sref
			, const string srefIxWdbeVPartype
			, const string titIxWdbeVPartype
			, const string stubRefWdbeMVector
			, const utinyint Length
			, const string Comment
		) {
	this->jnum = jnum;
	this->sref = sref;
	this->srefIxWdbeVPartype = srefIxWdbeVPartype;
	this->titIxWdbeVPartype = titIxWdbeVPartype;
	this->stubRefWdbeMVector = stubRefWdbeMVector;
	this->Length = Length;
	this->Comment = Comment;
};

bool WdbeQCmdARetpar::readXML(
			xmlXPathContext* docctx
			, string basexpath
			, bool addbasetag
		) {
	bool basefound;

	if (addbasetag)
		basefound = checkUclcXPaths(docctx, basexpath, basexpath, "WdbeQCmdARetpar");
	else
		basefound = checkXPath(docctx, basexpath);

	if (basefound) {
		extractStringUclc(docctx, basexpath, "sref", "srf", sref);
		extractStringUclc(docctx, basexpath, "srefIxWdbeVPartype", "pty", srefIxWdbeVPartype);
		extractStringUclc(docctx, basexpath, "titIxWdbeVPartype", "pty2", titIxWdbeVPartype);
		extractStringUclc(docctx, basexpath, "stubRefWdbeMVector", "vec", stubRefWdbeMVector);
		extractUtinyintUclc(docctx, basexpath, "Length", "len", Length);
		extractStringUclc(docctx, basexpath, "Comment", "cmt", Comment);
	};

	return basefound;
};

/******************************************************************************
 class ListWdbeQCmdARetpar
 ******************************************************************************/

ListWdbeQCmdARetpar::ListWdbeQCmdARetpar() {
};

ListWdbeQCmdARetpar::ListWdbeQCmdARetpar(
			const ListWdbeQCmdARetpar& src
		) {
	nodes.resize(src.nodes.size(), NULL);
	for (unsigned int i = 0; i < nodes.size(); i++) nodes[i] = new WdbeQCmdARetpar(*(src.nodes[i]));
};

ListWdbeQCmdARetpar& ListWdbeQCmdARetpar::operator=(
			const ListWdbeQCmdARetpar& src
		) {
	nodes.resize(src.nodes.size(), NULL);
	for (unsigned int i = 0; i < nodes.size(); i++) nodes[i] = new WdbeQCmdARetpar(*(src.nodes[i]));

	return *this;
};

ListWdbeQCmdARetpar::~ListWdbeQCmdARetpar() {
	clear();
};

void ListWdbeQCmdARetpar::clear() {
	for (unsigned int i = 0; i < nodes.size(); i++) if (nodes[i]) delete nodes[i];
	nodes.resize(0);
};

bool ListWdbeQCmdARetpar::readXML(
			xmlXPathContext* docctx
			, string basexpath
			, bool addbasetag
		) {
	bool basefound;

	vector<unsigned int> ics;
	vector<bool> shorttags;

	WdbeQCmdARetpar* rec = NULL;

	string s;

	if (addbasetag)
		basefound = checkUclcXPaths(docctx, basexpath, basexpath, "ListWdbeQCmdARetpar");
	else
		basefound = checkXPath(docctx, basexpath);

	extractList(docctx, basexpath, "WdbeQCmdARetpar", "row", "jnum", ics, shorttags);

	clear();

	for (unsigned int i = 0; i < ics.size(); i++) {
		rec = new WdbeQCmdARetpar(ics[i]);

		if (shorttags[i]) s = basexpath + "/row[@jnum='" + to_string(ics[i]) + "']";
		else s = basexpath + "/WdbeQCmdARetpar[@jnum='" + to_string(ics[i]) + "']";

		if (rec->readXML(docctx, s)) nodes.push_back(rec);
		else delete rec;
	};

	return basefound;
};

File: japiwdbe/WdbeQCmdARetpar.java
/**
	* \file WdbeQCmdARetpar.h
	* Java API code for record of table TblWdbeQCmdARetpar
	* \copyright (C) 2018-2020 MPSI Technologies GmbH
	* \author Alexander Wirthmueller (auto-generation)
	* \date created: 5 Dec 2020
  */
// IP header --- ABOVE

package apiwdbe;

import org.w3c.dom.*;
import sbecore.*;

public class WdbeQCmdARetpar {
	
	public WdbeQCmdARetpar(
				int jnum
				, String sref
				, String srefIxWdbeVPartype
				, String titIxWdbeVPartype
				, String stubRefWdbeMVector
				, int Length
				, String Comment
			) {
		this.jnum = jnum;
		this.sref = sref;
		this.srefIxWdbeVPartype = srefIxWdbeVPartype;
		this.titIxWdbeVPartype = titIxWdbeVPartype;
		this.stubRefWdbeMVector = stubRefWdbeMVector;
		this.Length = Length;
		this.Comment = Comment;
	};

	public int jnum;
	public String sref;
	public String srefIxWdbeVPartype;
	public String titIxWdbeVPartype;
	public String stubRefWdbeMVector;
	public int Length;
	public String Comment;
	
	public boolean readXML(
				Document doc
				, String basexpath
				, boolean addbasetag
			) {
		if (addbasetag) basexpath = Xmlio.checkUclcXPaths(doc, basexpath, "WdbeQCmdARetpar");

		if (Xmlio.checkXPath(doc, basexpath)) {
			sref = Xmlio.extractStringUclc(doc, basexpath, "sref", "srf", null, 0);
			srefIxWdbeVPartype = Xmlio.extractStringUclc(doc, basexpath, "srefIxWdbeVPartype", "pty", null, 0);
			titIxWdbeVPartype = Xmlio.extractStringUclc(doc, basexpath, "titIxWdbeVPartype", "pty2", null, 0);
			stubRefWdbeMVector = Xmlio.extractStringUclc(doc, basexpath, "stubRefWdbeMVector", "vec", null, 0);
			Length = Xmlio.extractIntegerUclc(doc, basexpath, "Length", "len", null, 0);
			Comment = Xmlio.extractStringUclc(doc, basexpath, "Comment", "cmt", null, 0);
			
			return true;
		};

		return false;
	};
};
